src/core/qgsprovidermetadata.h

```diff
@@ -249,6 +261,101 @@ class CORE_EXPORT QgsProviderMetadata
      */
     virtual QgsTransaction *createTransaction( const QString &connString ) SIP_FACTORY;
 
+    /**
+     * Returns a dictionary of provider connections,
+     * the dictionary key is the connection identifier.
+     * \param cached if FALSE connections will be re-read from the settings
+     * \since QGIS 3.10
+     */
+    virtual QMap<QString, QgsAbstractProviderConnection *> connections( bool cached = true );
```

Comment:
Ownership? Or factory?

Reply:
Ownership remains to the metadata class.

Reply:
I added it to dox